infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/secure/jwtproxy/JwtProxyConfigBuilderFactory.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.server.secure.jwtproxy;
+
+public interface JwtProxyConfigBuilderFactory {
```

Comment:
You can avoid having this factory by moving setting workspace id in JwtProxyConfigBuilder constructor to separate method. Please analyze this way.

Reply:
It is required field, so putting it in setter produces hidden condition (should always call setter) so i decided to leave it in constructor.

Reply:
Actually, it doesn't make sense to set up jwtproxy without invocation `#addVerifierProxy` methods, so it's also kind of required field. Maybe adding the corresponding check into `#build` method will help to avoid error because of this hidden condition.

Workspace id parameter is put into Audience field of Verifier proxy. It's why you can put workspace id parameter into `addVerifierProxy` method.
If you found that the following example 
```
proxyConfigBuilder.addVerifierProxy(workspaceId, listenPort, "http://" + backendServiceName + ":" + backendServicePort, excludes)
```
has too much parameters, you can consider adding builder
```
proxyConfigBuilder.addNewVerifierProxy()
                               .withWorkspaceId(workspaceId)
                               .withListenPort(listenPort)
                               .withUpstream("http://" + backendServiceName + ":" + backendServicePort)
                               .withExcludes(excludes)
                               .end();
```

I don't think that factory leads to a big issue, I'm OK to leave as it is. There are just thoughts.